api/envoy/api/v2/route/route.proto

```diff
@@ -708,49 +708,8 @@ message RouteAction {
   // ignoring the rest of the hash policy list.
   repeated HashPolicy hash_policy = 15;
 
-  // Indicates that a HTTP/1.1 client connection to this particular route is allowed to
-  // upgrade to a WebSocket connection. The default is false.
-  //
-  // .. attention::
-  //
-  //   If a connection is upgraded to a WebSocket connection, Envoy will set up plain TCP
-  //   proxying between the client and the upstream server. Hence, an upstream
-  //   server that rejects the WebSocket upgrade request is also responsible for
-  //   closing the associated connection. Until then, Envoy will continue to
-  //   proxy data from the client to the upstream server.
-  //
-  //   Redirects are not supported on routes where WebSocket upgrades are allowed.
-  google.protobuf.BoolValue use_websocket = 16 [deprecated = true];
-
-  message WebSocketProxyConfig {
-    // See :ref:`stat_prefix
-    // <envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.stat_prefix>`. If the parameter
-    // is not specified, the default value of "websocket" is used.
-    //
-    // WebSocket connections support the :ref:`downstream statistics
-    // <config_network_filters_tcp_proxy_stats>` for TCP proxy, except for the following, which are
-    // reported in the :ref:`HTTP Connection Manager statistics <config_http_conn_man_stats>`:
-    // - downstream_cx_tx_bytes_total
-    // - downstream_cx_tx_bytes_buffered
-    // - downstream_cx_rx_bytes_total
-    // - downstream_cx_rx_bytes_buffered
-    string stat_prefix = 1;
-
-    // See :ref:`idle_timeout
-    // <envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.idle_timeout>`. This timeout is
-    // only in effect after the WebSocket upgrade request is received by Envoy. It does not cover
-    // the initial part of the HTTP request.
-    google.protobuf.Duration idle_timeout = 2
```

Comment:
Does the idle timeout on the http conn manager, when doing websockets, behave similar-enough to this? Eg, when in websocket mode, it can't be based on requests, it should just be whether any traffic is flowing in either direction.